bundles/EcommerceFrameworkBundle/OrderManager/Order/Agent.php

```diff
@@ -163,15 +163,35 @@ public function itemChangeAmount(OrderItem $item, $amount)
         $oldAmount = $item->getAmount();
         $note->addData('amount.old', 'text', $oldAmount);
         $note->addData('amount.new', 'text', $amount);
-
+        $oldTotalPrice = $item->getTotalPrice();
+        $unitPrice = $oldTotalPrice/$oldAmount;
+        $newTotalPrice =$unitPrice*$amount;
         // change
         $item->setAmount($amount);
-
+        $item->setTotalPrice( $newTotalPrice);
```

Comment:
in that case we also would need to update netprice and tax information (if available)
![image](https://user-images.githubusercontent.com/8792145/89012200-11385d00-d312-11ea-88ab-42e4fea847a4.png)